---
title: "README"
author: "Yutian Liu"
date: "`r Sys.Date()`"
output: github_document
always_allow_html: true
---

# Research on the correlation between functional human genes and epigenetic conservation

## Mentor

Name: Dr. Kimberly Siegmund

Email: [kims\@usc.edu](mailto:kims@usc.edu){.email}

## Introduction

This project is based on the paper called ["Functional human genes typically exhibit epigenetic conservation"](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8439480/), where they have found that DepMap critical genes exhibit cell-specific preferential epigenetic conservation by comparing DNA methylation measurements across intestinal crypts from the same person as well as between duplicate cell lines.Therefore, they raised the hypothesis that essential genes are more conserved because if there is a deviation, it is not an important area of the genome since the body does not have to protect it.

## Things have done

-   Recauculated the relationship between Colon epithelial average gene expression (log2 CPM) and variability (variance) with gene conservation (PWD)
-   Reproduced and optimize drawn figure for a better visualization of the relationship

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r install libraries, message=FALSE,results = 'hide',include=FALSE}
list.of.packages <- c("readxl", "dplyr","ggpubr","ggstatsplot","cowplot","devtools","Matrix","data.table","ggridges")
new.packages <- list.of.packages[!(list.of.packages %in% installed.packages()[,"Package"])]
if(length(new.packages)) install.packages(new.packages)
```

```{r load libraries,message=FALSE}
library(readxl)
library(dplyr)
library(cowplot)
library(ggplot2)
library(ggpubr)
library(devtools)
library(Matrix)
library(ggridges)
library(data.table)
```

# Normal Colon data

Read Single cell RNA-seq data from normal colon (GSE125970 (38)).For normal colon, all rectal and colon cells were analyzed, and all epithelial cells were analyzed for the CRCs. Raw reads were converted to CPM (reads divided by total reads) and then to log2CPM (x+1). Average gene log2CPM expression and log2CPM variance were calculated for all cells, including zero values.

Colon epithelial average gene expression (log2 CPM) and variability (variance) can be calculated from single cell RNA-seq data.

Based on DepMap data [1] to identify “essential” genes.

```{r}
nmx <- readxl::read_xlsx("Normal_colon.xlsx")
```

```{r,include=FALSE}
essential_depmap<-nmx %>%
  filter(nmx$`common essential DepMap`!="#N/A")
Not_essential_depmap<-nmx[1220:19525,]
Nona_essential_depmap <- dplyr::filter(essential_depmap, !is.na(`ave NN3`))
Nona_NotDepMap <- dplyr::filter(Not_essential_depmap, !is.na(`ave NN3`))
Nona_NN <- dplyr::filter(nmx, !is.na(`ave NN3`))
```

```{r,include=FALSE}
sum(is.na(essential_depmap))
```

```{r, change column names}
change_col_name <- function(df){
  colnames(df)[2] <-"Log2_CPM"
  colnames(df)[3] <-"Log2_CPM_variance"
  colnames(df)[4] <-"NN3_genePWD"
  return(df)
}
dfs <- list(Nona_NN)
Nona_NN <-lapply(dfs, change_col_name )
Nona_NN <- rbindlist(Nona_NN)
dfs <- list(Nona_essential_depmap)
Nona_essential_depmap <-lapply(dfs, change_col_name )
Nona_essential_depmap <- rbindlist(Nona_essential_depmap)
dfs <- list(Nona_NotDepMap)
Nona_NotDepMap <-lapply(dfs, change_col_name )
Nona_NotDepMap <- rbindlist(Nona_NotDepMap)
```

```{r,include=FALSE}
Nona_NN$`common essential DepMap`[!is.na(Nona_NN$`common essential DepMap`)] <- "DepMap_Essential"
Nona_NN$`common essential DepMap`[is.na(Nona_NN$`common essential DepMap`)] <- "Not_DepMap_Essential"
```

## DepMap Essential Genes

```{r}
cor(Nona_essential_depmap$NN3_genePWD, Nona_essential_depmap$Log2_CPM, method = c("pearson", "kendall", "spearman"))
cor.test(Nona_essential_depmap$NN3_genePWD, Nona_essential_depmap$Log2_CPM, method=c("pearson", "kendall", "spearman"))
```

If the correlation coefficient is greater than zero, it is a positive relationship. Conversely, if the value is less than zero, it is a negative relationship. A value of zero indicates that there is no relationship between the two variables. From the result, negative and significant correlations (p < 0.05) were observed between gene PWDs and gene expression/variability.

```{r}
cor.test(Nona_essential_depmap$NN3_genePWD, Nona_essential_depmap$Log2_CPM, alternative = "less")
```

### Visualization

#### Figures Reproduced

```{r,echo=FALSE}
ggplot(Nona_essential_depmap,aes(y=NN3_genePWD,x=Log2_CPM)) + geom_point(alpha = 0.3) + geom_smooth(method = "lm",color="red",formula = y~x) + stat_cor(method = "pearson", label.x = 0.02, label.y = 0.14,size=3)+ggtitle("Reproced expression Essential Genes: PWD_vs_Log2CPM")
```
```{r,echo=FALSE}
ggplot(Nona_essential_depmap,aes(y=NN3_genePWD,x=Log2_CPM_variance)) + geom_point(alpha = 0.3) + geom_smooth(method = "lm",color="red",formula = y~x) + stat_cor(method = "pearson", label.x = 0.02, label.y = 0.14,size=3)+ggtitle("Reproced variavility Essential Genes: PWD_vs_Log2CPM")
```

#### Density Plots & Change Y-X axis

```{r,echo=FALSE}
smoothScatter(Nona_essential_depmap$Log2_CPM~ Nona_essential_depmap$NN3_genePWD,main = "Essential_depmap_normal_colon: Log2CPM vs PWD",xlab = "gene conservation(PWD)",ylab = "Log2_CPM")
```
```{r,echo=FALSE}
# Bin size control + color palette
ggplot(Nona_essential_depmap, aes(x=NN3_genePWD, y=Log2_CPM) ) + xlab ("gene conservation(PWD)")+ylab( "Log2_CPM")+geom_bin2d() +
  scale_fill_continuous(type = "viridis") +
  theme_bw()
```

Many PWDs are concentrated in the range from 0.06 to 0.08 

```{r,echo=FALSE}
ggplot(Nona_essential_depmap,aes(x=NN3_genePWD,y=Log2_CPM)) + geom_point(alpha = 0.3) + geom_smooth(color="red",method ='lm',formula = y~x) + stat_cor(method = "pearson", label.x = 0.02, label.y = 8,size=3)+ggtitle("Expression Essential Genes: Log2CPM_vs_PWD")
```

```{r,echo=FALSE}
ggplot(Nona_essential_depmap,aes(x=NN3_genePWD,y=Log2_CPM_variance)) + geom_point(alpha = 0.3) + geom_smooth(color="red",method ='lm',formula = y~x) + stat_cor(method = "pearson", label.x = 0.02, label.y = 12,size=3)+ggtitle("Variance Essential Genes: Log2CPM_vs_PWD")
```

## Non-DepMap Essential Genes

```{r}
cor(Nona_NotDepMap$NN3_genePWD, Nona_NotDepMap$Log2_CPM, method = c("pearson", "kendall", "spearman"))
cor.test(Nona_NotDepMap$NN3_genePWD, Nona_NotDepMap$Log2_CPM, method=c("pearson", "kendall", "spearman"))
```

### Visualization

#### Figures Reproduced

```{r,echo=FALSE}
ggplot(Nona_NotDepMap,aes(y=NN3_genePWD,x=Log2_CPM)) + geom_point(alpha = 0.3) + geom_smooth(method = "lm",color="red",formula = y~x) + stat_cor(method = "pearson", label.x = 0.02, label.y = 0.4,size=3)+ggtitle("Reproced expression Non-Essential Genes: PWD_vs_Log2CPM")
```
```{r,echo=FALSE}
ggplot(Nona_NotDepMap,aes(y=NN3_genePWD,x=Log2_CPM_variance)) + geom_point(alpha = 0.3) + geom_smooth(method = "lm",color="red",formula = y~x) + stat_cor(method = "pearson", label.x = 0.02, label.y = 0.4,size=3)+ggtitle("Reproced variavility Essential Genes: PWD_vs_Log2CPM")
```

#### Density Plots & Change Y-X axis

```{r,echo=FALSE}
smoothScatter(Nona_NotDepMap$Log2_CPM~ Nona_NotDepMap$NN3_genePWD,main = "Non-Essential_depmap_normal_colon: Log2CPM vs PWD",xlab = "gene conservation(PWD)",ylab = "Log2_CPM")
```

```{r,echo=FALSE}
# Bin size control + color palette
ggplot(Nona_NotDepMap, aes(x=NN3_genePWD, y=Log2_CPM) ) + xlab ("gene conservation(PWD)")+ylab( "Log2_CPM")+geom_bin2d() +
  scale_fill_continuous(type = "viridis") +
  theme_bw()
```

Many PWDs are concentrated in the range from 0.05 to 0.15

```{r,echo=FALSE}
ggplot(Nona_NotDepMap,aes(x=NN3_genePWD,y=Log2_CPM)) + geom_point(alpha = 0.3) + geom_smooth(color="red",method ='lm',formula = y~x) + stat_cor(method = "pearson", label.x = 0.3, label.y = 8,size=3)+ggtitle("Non-Expression Essential Genes: Log2CPM_vs_PWD")
```

```{r,echo=FALSE}
ggplot(Nona_NotDepMap,aes(x=NN3_genePWD,y=Log2_CPM_variance)) + geom_point(alpha = 0.3) + geom_smooth(color="red",method ='lm',formula = y~x) + stat_cor(method = "pearson", label.x = 0.3, label.y = 12,size=3)+ggtitle("Non-Variance Essential Genes: Log2CPM_vs_PWD")
```

## Comparison between Essential genes & Non-Essential genes

```{r}
# Plot distributions
Nona_NN %>%
  ggplot(aes(x = Log2_CPM,
             y = `common essential DepMap`,
             fill = `common essential DepMap`)) +
  ggridges::geom_density_ridges(bandwidth = 4) + ggtitle("Essential, Non-essential gene expression comparison ")
```
```{r}
# Compute the mean for and size of each group
group_means <-
  Nona_NN %>%
  group_by(`common essential DepMap`) %>%
  summarise(mean = mean(Log2_CPM),
            n = n())

group_means
```

```{r}
# Create our data visualisation
Nona_NN %>%
  ggplot(aes(x = `common essential DepMap`, y = Log2_CPM, fill = `common essential DepMap`)) +
  geom_boxplot() +

  # Add the mean for each group
  geom_point(data = group_means,
             aes(x = `common essential DepMap`, y = mean),
             shape = 3,
             size = 2)
```
```{r}
# Compute the mean for and size of each group
group_means <-
  Nona_NN %>%
  group_by(`common essential DepMap`) %>%
  summarise(mean = mean(NN3_genePWD),
            n = n())
# Create our data visualisation
Nona_NN %>%
  ggplot(aes(x = `common essential DepMap`, y = NN3_genePWD, fill = `common essential DepMap`)) +
  geom_boxplot() +

  # Add the mean for each group
  geom_point(data = group_means,
             aes(x = `common essential DepMap`, y = mean),
             shape = 3,
             size = 2)
```

## Preliminary Conclusion

More variably and highly expressed genes are more conserved in normal and neoplastic colon.

Colon epithelial average gene expression (log2 CPM) and variability (variance) can be calculated from single cell RNA-seq data and correlated (Pearson coefficient values in parenthesis) with gene conservation (PWD). Negative and significant correlations (p < 0.05) were observed between gene PWDs and gene expression/variability.

PWDs in DepMap essential genes are concentrated in the range from 0.06 to 0.08 and PWDs in Non-DepMap essential gene are concentrated in the range from 0.05 to 0.15.

## 1.20 Update Interactive Plots about TFs in paper



In the paper "Transcription factor expression as a predictor of colon cancer prognosis: a machine learning practice"They identified five transcription factors for the predictive model, and they are HOXC9, ZNF556, HEYL, HOXC4 and HOXC6. Those five TFs(transcription factors) are not in the common essential genes list. And in the paper "DNA methylation events in transcription factors and gene expression changes in colon cancer", they selected 44 m-genes (methylation associated genes) which showed a partial correlation adjusting by stromal content. Those interactive plots are saved in this link. [Interactive Plots](https://yutianl.shinyapps.io/DNAMethylationScatterPl/) You can chose from the right legend and click the button twice to isolate them for a better visualization. Choose the `Log2_CPM` as the Y variable and `NN3_genePWD` as the X variable to view the association between gene expression and pair distance.



## Things to do

-   Download the transcription factor binding sites from [JASPAR database](https://jaspar.genereg.net/) to compare the regions we found interesting to examine if this hypothesis still stands in terms of transcriptome markers.
